Sources/AppCoinsSDK/Models/Services/Billing/AppCoinBillingClient.swift

```diff
@@ -553,32 +553,44 @@ internal class AppCoinBillingClient : AppCoinBillingService {
         }
     }
     
-    internal func getSupportedCurrencies(result: @escaping (Result<CurrencyListRaw, BillingError>) -> Void) {
-        if let requestURL = URL(string: "\(endpoint)/currencies?type=FIAT&icon.height=128") {
-            var request = URLRequest(url: requestURL)
-            
-            request.httpMethod = "GET"
-            request.addValue("application/json", forHTTPHeaderField: "Content-Type")
-            request.timeoutInterval = 10
-            
-            let task = URLSession.shared.dataTask(with: request) { data, response, error in
+    internal func getSupportedCurrencies(result: @escaping (Result<[CurrencyRaw], BillingError>) -> Void) {
+        var allCurrencies: [CurrencyRaw] = []
+        
+        func getAllCurrencies(url: URL? = nil) {
```

Comment:
I really like this solution. Well done. 

Just three tiny details I would improve:
1. I would build the first URL outside the getAllCurrencies() method and make the url parameter non-optional.
2. Call the `allCurrencies` variable simply `currencies` – no need to add complexity to the variable names.
3. Call the `getAllCurrencies` method `getNextCurrenciesBatch` – this is picky but what we're doing here is recursively getting batches of currencies, not getting them all as the method name indicates.

These are all really small details and are suggestions. The solution is OK and I will not reject the PR if you decide to move forward like this.